src/runtime/arc.ts

```diff
@@ -105,7 +105,10 @@ constructor({id, context, pecFactories, slotComposer, loader, storageKey, storag
     this._context = context || new Manifest({id});
     // TODO: pecFactories should not be optional. update all callers and fix here.
     this.pecFactories = pecFactories && pecFactories.length > 0 ? pecFactories.slice() : [FakePecFactory(loader).bind(null)];
-
+    // TODO(sjmiles): FIXME: currently some UiBrokers need to recover arc from composer in order to forward events
+    if (slotComposer && !slotComposer['arc']) {
```

Comment:
this is a workaround that was removed in:
https://github.com/PolymerLabs/arcs/pull/3506/commits/cb9a48e5046cc735c2fc3993521517b59307856c
did you intentionally add it back as part of this PR?

Reply:
Yes, because the workaround turned out to still be necessary as per the TODO comment.